### PR TITLE
Data source version endpoints

### PR DESCRIPTION
**Pull request by AustinSMueller**

### CODE REVIEW

**Comment by AustinSMueller** [comment]
- Add OpenSearch status endpoint (/opensearch-version)
- Add Neo4j status endpoint (/neo4j-version)
- Minor GraphQLController cleanup
- Remove and replace APIError classes

File: controller/GraphQLController.java [hunk 1]
```diff
@@ -50,55 +56,116 @@ public GraphQLController(ConfigurationDAO config, BentoGraphQL bentoGraphQL){
...
+				return responseEntity;
+			}
+			//Parse version from response
+			String version = gson.fromJson(responseEntity.getBody(), JsonObject.class)
+					.get("data").getAsJsonObject()
+					.get("neo4jVersion").getAsString();
+			//Format and return info
+			return ResponseEntity.ok(gson.toJson(new DataSourceVersion(uri, version)));
+		}
+		catch (Exception e){
```
Comment (knockknockyoo) [inline controller/GraphQLController.java]: Optional suggestion: try and catch block duplicates; use strategy design pattern.

File: controller/GraphQLController.java [hunk 2]
```diff
@@ -50,55 +56,116 @@ public GraphQLController(ConfigurationDAO config, BentoGraphQL bentoGraphQL){
...
+					.get("version").getAsJsonObject()
+					.get("number").getAsString();
+			//Format and return info
+			return ResponseEntity.ok(gson.toJson(new DataSourceVersion(uri, versionString)));
+		}
+		catch (IOException|InterruptedException e) {
+			return logAndReturnError(HttpStatus.NOT_FOUND,
+					"The OpenSearch service could not be reached", e);
+		}
+		catch (Exception e){
```
Comment (knockknockyoo) [inline controller/GraphQLController.java]: Optional suggestion: try and catch block duplicates; use strategy design pattern.

File: controller/GraphQLController.java [hunk 3]
```diff
@@ -50,55 +56,116 @@ public GraphQLController(ConfigurationDAO config, BentoGraphQL bentoGraphQL){
...
+
+	@CrossOrigin
+	@RequestMapping(value = "/opensearch-version", method = {RequestMethod.GET},
+			produces = MediaType.APPLICATION_JSON_VALUE + "; charset=utf-8")
+	public ResponseEntity<String> getOpenSearchVersion(){
+		logger.info("Hit end point: /opensearch-version");
+		try{
+			//Build URI
+			URIBuilder builder = new URIBuilder()
+					.setScheme("http")
```
Comment (knockknockyoo) [inline controller/GraphQLController.java]: This is probably not working on AWS. Please, check whether AWS is using https. Or, you could utilize EsService.send(request) instead.

**Review by knockknockyoo — COMMENTED** [review]
It looks good. Please take a look my suggestions. Thanks!

**Comment by knockknockyoo** [comment]
Thank you for taking your time.  It looks very nice :)

**Pull request merged**